packages/server/lib/StudioLifecycleManager.ts

```diff
@@ -179,4 +201,35 @@ export class StudioLifecycleManager {
       this.listeners.push(listener)
     }
   }
+
+  private updateStatus (status: StudioStatus) {
+    if (status === this.lastStatus) return
+
+    debug('Studio status changed: %s → %s', this.lastStatus, status)
+    this.lastStatus = status
+
+    if (this.ctx) {
+      process.nextTick(() => {
+        this.ctx?.emitter.studioStatusChange()
+      })
+    } else {
+      debug('No ctx available, cannot emit studioStatusChange')
+    }
+  }
+
+  // Monitor status changes on the studioManager
+  private setupStatusProxy (studioManager: StudioManager) {
+    let currentStatus = studioManager.status
+
+    Object.defineProperty(studioManager, 'status', {
```

Comment:
Don't we control the studio manager? Why do we need to override the `status` property?

Reply:
I'm not sure what you mean - the goal here is to emit an event any time the `status` property changes

Reply:
Yeah, I was thinking we'd come up with some other pattern to get the event emitted. Since, we control the `StudioManager`, we shouldn't need to redefine the `status` property here.

Reply:
updated this to make it so that studio manager exposes a method to update the status instead of watching the status property